packages/graphql/src/services/notifications.ts

```diff
@@ -49,3 +51,130 @@ export async function subscribeToPush(
     },
   })
 }
+
+interface UnsubscribeToPushArgs {
+  courseId: string
+  endpoint: string
+}
+
+export async function unsubscribeFromPush( 
+  { courseId, endpoint }: UnsubscribeToPushArgs,
+  ctx: ContextWithUser
+) {
+  try {
+    await ctx.prisma.pushSubscription.delete({
+      where: {
+        participantId_courseId_endpoint: {
+          participantId: ctx.user.sub,
+          courseId,
+          endpoint
+        }
+      }
+    })
+    return true
+  } catch(error) {
+    console.error("An error occured while trying to unsubscribe from push notifications: ", error)
+    return false
+  }
+}
+
+export async function sendPushNotifications( ctx: Context) {
+  webpush.setVapidDetails(
```

Comment:
before doing anything, compare the secret passed as an argument and compare it to a secret token from process.env (e.g., the PUSH\_SECRET)

Reply:
also, when adding a variable to process.env, also remember to add it to doppler, as well as to the deployment code (in deploy/) -> ask RS for specifics on this